aten/src/ATen/core/jit_type.h

```diff
@@ -1847,12 +1873,23 @@ struct CAFFE2_API ClassType : public NamedType {
   }
 
   bool is_module() const override {
-    return bool(parameterSlots_);
+    return isModule_;
+  }
+
+  const std::vector<AttributeKind>& getAllAttributes() const {
+    return attributes_;
   }
+
   bool is_parameter(size_t slot) const {
     TORCH_INTERNAL_ASSERT(
         is_module(), "asking for parameterSlots of non-Module");
-    return parameterSlots_->at(slot);
+    return attributes_.at(slot).getKind() == ATTRIBUTE_KIND_ENUM::PARAMETER;
+  }
+
+  bool is_buffer_written_attribute(size_t slot) const {
```

Comment:
I'm seeing a few different ways of wording "is this a buffer?". Here it's `is_buffer_written_attribute`, and in `addAttribute` it's `wasRegisteredAsBuffer`. I think similar to parameters, it should just be `is_buffer` everywhere.

Reply:
One is a method, so it asks the question, and another is a param, so it tells the function. is_buffer works for both though!